java/src/org/openqa/selenium/grid/server/ReverseProxyFlags.java

```diff
@@ -0,0 +1,45 @@
+// Licensed to the Software Freedom Conservancy (SFC) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The SFC licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.openqa.selenium.grid.server;
+
+import static org.openqa.selenium.grid.config.StandardGridRoles.ROUTER_ROLE;
+import static org.openqa.selenium.grid.server.ReverseProxyOptions.REVERSE_PROXY_SECTION;
+
+import com.beust.jcommander.Parameter;
+import com.google.auto.service.AutoService;
+import java.util.Collections;
+import java.util.Set;
+import org.openqa.selenium.grid.config.ConfigValue;
+import org.openqa.selenium.grid.config.HasRoles;
+import org.openqa.selenium.grid.config.Role;
+
+@AutoService(HasRoles.class)
+public class ReverseProxyFlags implements HasRoles {
+
+  @Parameter(
+    names = {"--sub-path"},
+    arity = 1,
+    description = "A sub-path that should be considered for all user facing routes on the Hub/Router/Standalone")
+  @ConfigValue(section = REVERSE_PROXY_SECTION, name = "sub-path", example = "my_company/selenium_grid")
+  public String subPath;
+
+  @Override
+  public Set<Role> getRoles() {
+    return Collections.singleton(ROUTER_ROLE);
+  }
```

Comment:
This can be part of the `RouterFlags`, no need for a new class.

Reply:
Doing this would basically mean that I would need to alter dependencies because it looks like both `Hub` and `Standalone` don't have this package exposed to them via bazel. Is that fine @diemol

Reply:
Am seeing this when running the medium-tests from command line

```
# Execution platform: @local_config_platform//:host
java/src/org/openqa/selenium/grid/commands/Hub.java:44: error: package org.openqa.selenium.grid.router.httpd does not exist
import org.openqa.selenium.grid.router.httpd.RouterOptions;
```

I guess bazel is not happy here ?

Reply:
Please ignore this. Figured out what needs to be done.